src/transforms/workflow.ts

```diff
@@ -33,42 +37,91 @@ function mapStatus(status: string, conclusion?: string): string {
 	}
 }
 
-function mapPullRequests(pull_requests: GitHubPullRequest[]): JiraPullRequest[] {
-	return pull_requests.map(pr => (
-		{
-			commit: {
-				id: pr.head.sha,
-				repositoryUri: pr.head.repo.url,
-			},
-			ref: {
-				name: pr.head.ref,
-				uri: `${pr.head.repo.url}/tree/${pr.head.ref}`,
-			},
-		}
-	));
+function mapPullRequests(
+	pull_requests: GitHubPullRequest[]
+): JiraPullRequest[] {
+	return pull_requests.map((pr) => ({
+		commit: {
+			id: pr.head.sha,
+			repositoryUri: pr.head.repo.url,
+		},
+		ref: {
+			name: pr.head.ref,
+			uri: `${pr.head.repo.url}/tree/${pr.head.ref}`,
+		},
+	}));
 }
 
-export default (context: Context): JiraBuildData | undefined => {
-	const { workflow_run, workflow } = context.payload;
-	const issueKeys = issueKeyParser().parse(`${workflow_run.head_branch}\n${workflow_run.head_commit.message}`);
+export default async (
+	githubClient: GitHubAPI,
+	payload: WorkflowPayload,
+	jiraHost: string,
+	logger?: LoggerWithTarget
```

Comment:
Would it make sense to make it mandatory? Seems it is passed in all calls... (For tests we could always pass getLogger('test')

Reply:
And I'm already doing that 😄. Will get rid of the ?